File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/home/viewmodel/RobotsViewModelFactory.java
package com.marcosvbras.robomarket.flows.home.viewmodel;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.marcosvbras.robomarket.interfaces.BaseActivityCallback;

import javax.inject.Inject;

public class RobotsViewModelFactory implements ViewModelProvider.Factory {

    private final BaseActivityCallback activityCallback;

    @Inject
    public RobotsViewModelFactory(@NonNull BaseActivityCallback activityCallback) {
        this.activityCallback = activityCallback;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new RobotsViewModel(activityCallback);
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/home/viewmodel/RobotsViewModelFactory.kt
package com.marcosvbras.robomarket.flows.home.viewmodel

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.marcosvbras.robomarket.flows.home.interfaces.HomeActivityCallback
import com.marcosvbras.robomarket.interfaces.ActivityCallback

class RobotsViewModelFactory(private val activityCallback: HomeActivityCallback) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return RobotsViewModel(activityCallback) as T
    }

}